### ISSUE TITLE
Unable to load TTF/OTF files from map file with LuaHBTeX ("cannot open font file for reading")

### ISSUE THREAD

**Issue by diabonas** [latex3/luaotfload#142]
Consider the following plain TeX example trying to load the [Cuprum](https://ctan.org/pkg/cuprum) font:
```tex
%\input luaotfload.sty
\font\cuprum=cprmn8t
\cuprum
a
\bye
```
This document compiles without issues with both LuaTeX and LuaHBTeX:
```
This is LuaHBTeX, Version 1.12.0 (TeX Live 2020/Arch Linux)
 restricted system commands enabled.
(./mwe.tex [1{/var/lib/texmf/fonts/map/pdftex/updmap/pdftex.map}]){/usr/share/t
exmf-dist/fonts/enc/ttf2pk/base/T1-WGL4.enc}</usr/share/texmf-dist/fonts/truety
pe/public/cuprum/Cuprum-Regular.ttf></usr/share/texmf-dist/fonts/type1/public/a
msfonts/cm/cmr10.pfb>
Output written on mwe.pdf (1 page, 10127 bytes).
Transcript written on mwe.log.
```
Now uncomment the first line to load luaotfload and try compiling the file again. This works fine with LuaTeX, but LuaHBTeX now fails with a fatal error:
```
This is LuaHBTeX, Version 1.12.0 (TeX Live 2020/Arch Linux)
 restricted system commands enabled.
(./mwe.tex (/usr/share/texmf-dist/tex/luatex/luaotfload/luaotfload.sty
(/usr/share/texmf-dist/tex/latex/base/ltluatex.tex)) [1{/var/lib/texmf/fonts/ma
p/pdftex/updmap/pdftex.map}]){/usr/share/texmf-dist/fonts/enc/ttf2pk/base/T1-WG
L4.enc}
! error:  (file Cuprum-Regular.ttf) (ttf font): cannot open font file for readi
ng 'Cuprum-Regular.ttf'
!  ==> Fatal error occurred, no output PDF file produced!
```
This problem appears to affect all TTF/OTF fonts that are loaded from a map file, [`cuprum.map`](http://mirrors.ctan.org/fonts/cuprum/map/cuprum.map) in this case. Loading Type 1 fonts (e.g. `\font\nimbusroman=rtxptmr`) works without issues, as does using the TTF file name of the font (`\font\cuprum=Cuprum-Regular.ttf`). Since the font loads fine when commenting the first line, the problem doesn't seem to be with LuaHBTeX itself, but in the luaotfload code.

The issue affects e.g. the `adobe-garamond` option of the [mathdesign](https://ctan.org/pkg/mathdesign) package, see the downstream Arch Linux bug report [FS#66139](https://bugs.archlinux.org/task/66139).

**Comment by diabonas** [comment]
I ought to have mentioned that I am using the most recent luaotfload version 3.12. I also tried the current state of the dev branch (commit 79b7fdb1bda7b1ee912105281cb212dc677c7187), the issue can be reproduced there as well.

**Comment by diabonas** [comment]
A proposed fix is in #143.
